src/Raven.Server/Documents/TcpHandlers/SubscriptionDocumentsFetcher.cs

```diff
@@ -133,51 +139,24 @@ public class SubscriptionDocumentsFetcher
                         }
                     }
 
-                    if (++size >= _maxBatchSize)
+                    if (++numberOfDocs >= _maxBatchSize)
                         yield break;
 
-                    if (CanContinueBatch(docsContext) == false)
+                    if (size >= _MaximumAllowedMemory)
                         yield break;
-
                 }
             }
         }
 
-        public bool CanContinueBatch(DocumentsOperationContext ctx)
-        {
-            var totalAllocated = _threadAllocations.TotalAllocated;
-            _threadAllocations.CurrentlyAllocatedForProcessing = totalAllocated;
-            var currentlyInUse = new Size(totalAllocated, SizeUnit.Bytes);
-            if (currentlyInUse > _currentMaximumAllowedMemory)
-            {
-                if (MemoryUsageGuard.TryIncreasingMemoryUsageForThread(_threadAllocations, ref _currentMaximumAllowedMemory,
-                        currentlyInUse,
-                        _db.DocumentsStorage.Environment.Options.RunningOn32Bits, _logger, out var memoryUsage) ) return true;
-                var reason = $"Stopping the batch because cannot budget additional memory. Current budget: {currentlyInUse}.";
-                if (memoryUsage != null)
-                {
-                    reason += " Current memory usage: " +
-                              $"{nameof(memoryUsage.WorkingSet)} = {memoryUsage.WorkingSet}," +
-                              $"{nameof(memoryUsage.PrivateMemory)} = {memoryUsage.PrivateMemory}";
-                }
-
-                if(_logger.IsInfoEnabled)
-                    _logger.Info(reason);
-
-                ctx.DoNotReuse = true;
-
-                return false;
-
-            }
-            return true;
-        }
-
         private IEnumerable<(Document Doc, Exception Exception)> GetRevisionsToSend(
             DocumentsOperationContext docsContext,
             IncludeDocumentsCommand includesCmd,
             long startEtag)
         {
-            int size = 0;
+            int numberOfDocs = 0;
+            Size size = new Size(0, SizeUnit.Megabytes);
+            _MaximumAllowedMemory = new Size((PlatformDetails.Is32Bits ||
```

Comment:
This should be initialized once, not each time

Reply:
done